src/algebra/ring.lean

```diff
@@ -312,7 +312,18 @@ instance {α : Type*} {β : Type*} [semiring α] [semiring β] : has_coe (α →
 
 namespace ring_hom
 
-variables {β : Type v} [semiring α] [semiring β]
+/-- Interpret `f : α → β` with `is_semiring_hom f` as a ring homomorphism. -/
+def of {α : Type u} {β : Type v} [semiring α] [semiring β]
+  (f : α → β) [is_semiring_hom f] : α →+* β :=
+{ to_fun := f,
+  .. monoid_hom.of f,
+  .. add_monoid_hom.of f }
+
+@[simp] lemma coe_of {α : Type u} {β : Type v} [semiring α] [semiring β]
+  (f : α → β) [is_semiring_hom f] : ⇑(of f) = f := rfl
+
+variables {β : Type v} {γ : Type w} [semiring α] [semiring β] [semiring γ]
```

Comment:
Why don't you move this line up a bit? Could give you a cleaner statement of `of` and `coe_of`.

Reply:
Done.